tests/rptest/services/redpanda_cloud.py

```diff
@@ -834,12 +840,15 @@ def _select_cluster_id(self):
             # if there is still no id, just copy what globals.json had
             return self.config.id
 
-    def create(self, superuser: Optional[SaslCredentials] = None) -> str:
+    def create(self, superuser: SaslCredentials) -> str:
```

Comment:
We never supported `None` superuser here (we'd throw immediately since we access it w/o checking if it's null) so just remove the optionality which simplifies a lot of downstream code.